azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1900,9 +1923,22 @@ func flattenKubernetesClusterManagedClusterIdentity(input *containerservice.Mana
 		identity["tenant_id"] = *input.TenantID
 	}
 
+	identity["user_assigned_identity_id"] = ""
+	if input.UserAssignedIdentities != nil {
+		keys := []string{}
+		for key := range input.UserAssignedIdentities {
+			keys = append(keys, key)
+		}
+		parsedId, err := msiparse.UserAssignedIdentityID(keys[0])
```

Comment:
there's a panic here if `input.UserAssignedIdentities` is a non-nil but empty array, so could we make this:

```suggestion
		if len(keys) > 0 {
			parsedId, err := msiparse.UserAssignedIdentityID(keys[0])
			if err != nil {
 				return nil, err
 			}
	 		identity["user_assigned_identity_id"] = parsedId.ID()
		}
```